test/basic_routing/main.tf

```diff
@@ -0,0 +1,111 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE A ROUTE53 ZONE WITH SUBDOMAINS AND CNAMES
+# This example creates a zone and records for the main domain and a subdomain.
+#   - (www.)acme.com
+#   - (www.)dev.acme.com
+#
+# The www. subdomains are implement through CNAMES and point on the A records.
+# ---------------------------------------------------------------------------------------------------------------------
+
+# ---------------------------------------------------------------------------------------------------------------------
+# Configure the AWS Provider
+# ---------------------------------------------------------------------------------------------------------------------
+
+provider "aws" {
+  region = var.aws_region
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# Create a public s3 bucket that will act as a website
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "aws_s3_bucket" "website" {
```

Comment:
I would strongly **vote against** this. 

The scope of this test is to test a unit. If the unit has dependencies we should mock it ( which is kind of what we are doing ).  In our base modules, we should **never** cause dependencies to other modules since the base modules shouldn't cause any interdependencies.

Integrations we will test in modules that actually use more than one base module.

Reply:
> adding a bucket makes it an integration test as you need extra resources...
> 
> i am fine with this for now.. but we should actually test ist against our s3 module in this case and see how those play together...

Thought about it once again. Definitely not a fan of causing dependencies between base modules. This will cause a lot of additional overhead and modules should run without having any dependencies on other modules. Would strongly vote for leaving things as it is and use plain resources instead.

Also, this was already merged in another PR and shouldn't be part of this PR after all. Feel free to open a ticket or discussion for later if you'd like to change this.